File: cpp/Array/L_findAllDuplicateInArr.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> findDuplicates(vector<int> &nums)
{
    vector<int> result;
    for (int i = 0; i < nums.size(); i++)
    {
        int num = abs(nums[i]);
        if (nums[num - 1] < 0)
            result.push_back(num);
        nums[num - 1] = nums[num - 1] * -1;
    }
    return result;
}

File: Python/Arrays/L_findAllDuplicateInArr.py
from typing import List


def findDuplicates(nums: List[int]) -> List[int]:
    result = []
    for n in nums:
        n = abs(n)
        if nums[n-1] < 0:
            result.append(n)
        nums[n-1] = -nums[n-1]
    return result